auth/service.go

```diff
@@ -170,10 +170,21 @@ func (svc service) Identify(ctx context.Context, token string) (Identity, error)
 	}
 }
 
-func (svc service) Authorize(ctx context.Context, token, sub, obj, act string) (bool, error) {
+func (svc service) Authorize(ctx context.Context, subject, object, relation string) (bool, error) {
+	ar, err := svc.policyAgent.CheckPolicy(ctx, subject, object, relation)
+	if err != nil {
+		return false, errors.Wrap(ErrAuthorization, err)
+	}
+	if ar.AuthzError != nil {
+		return false, errors.Wrap(ErrAuthorization, err)
+	}
 	return true, nil
 }
 
+func (svc service) AddPolicy(ctx context.Context, subject, object, relation string) error {
+	return svc.policyAgent.AddPolicy(ctx, subject, object, relation)
```

Comment:
Is there are other agents also? Like Auth agent or identity agent? If no, maybe just an agent makes more sense? 
Like svc.agent.addPolicy ? If its only one agent , method name is already self-documented

Reply:
Under `auth` package, there is no other service called `agent`. Auth Service has `Authn`, `Authz`, and `GroupService`. I will rename `policyAgent` to `agent`